src/clj/fluree/db.cljc

```diff
@@ -0,0 +1,521 @@
+  alias."
+  [ledger-alias-or-address]
+  (jld-ledger/fluree-address? ledger-alias-or-address))
+
+(defn create
+  "Creates a new json-ld ledger. A connection (conn)
+  must always be supplied.
+
+  Ledger-name (optional) is a friendly name that is used for:
+  - When publishing to a naming service that allows multiple pointers for the
+    same namespace (e.g. IPNS), this becomes a sub-directory off the namespace.
+    For multple directories deep, use '/' for a
+    e.g. the ledgers movies/popular, books/authors, books/best-sellers could
+    use the same IPNS id (in this example using IPNS DNSLink):
+    fluree:ipns://my.dns.com/books/authors
+    fluree:ipns://my.dns.com/books/best-sellers
+    fluree:ipns://my.dns.com/movies/top-rated
+  - When combining multiple ledgers, each ledger becomes an individual named
+    graph which can be referenced by name.
+
+  Options map (opts) can include:
+  - did - DId information to use, if storing blocks as verifiable credentials"
+  ([conn] (create conn nil nil))
+  ([conn ledger-alias] (create conn ledger-alias nil))
+  ([conn ledger-alias opts]
+   (promise-wrap
+    (do
+      (log/info "Creating ledger" ledger-alias)
+      (jld-ledger/create conn ledger-alias opts)))))
+
+(defn alias->address
+  "Returns a core.async channel with the connection-specific address of the
+  given ledger-alias."
+  [conn ledger-alias]
+  (log/debug "Looking up address for ledger alias" ledger-alias)
+  (nameservice/primary-address conn ledger-alias nil))
+
+(defn load
+  "Loads an existing ledger by its alias (which will be converted to a
+  connection-specific address first)."
+  [conn alias-or-address]
+  (promise-wrap
+    (jld-ledger/load conn alias-or-address)))
+
+(defn exists?
+  "Returns a promise with true if the ledger alias or address exists, false
+  otherwise."
+  [conn ledger-alias-or-address]
+  (promise-wrap
+    (go
+      (let [address (if (address? ledger-alias-or-address)
+                      ledger-alias-or-address
+                      (<! (alias->address conn ledger-alias-or-address)))]
+        (log/debug "exists? - ledger address:" address)
+        (<! (nameservice/exists? conn address))))))
+
+(defn notify
+  "Notifies the connection with a new commit map (parsed JSON commit with string keys).
+
+  If the connection knows of the ledger, and is currently maintaining
+  an in-memory version of the ledger, will attempt to update the db if the commit
+  is for the next 't' value. If a commit is for a past 't' value, noop.
+  If commit is for a future 't' value, will drop in-memory ledger for reload upon next request."
+  [conn commit-map]
+  (promise-wrap
+    (if (map? commit-map)
+      (notify-ledger conn commit-map)
+      (go
+        (ex-info (str "Invalid commit map, perhaps it is JSON that needs to be parsed first?: " commit-map)
+                 {:status 400 :error :db/invalid-commit-map})))))
+
+
+(defn index
```

Comment:
Should we remove all of these stubs? I don't know that we even expect to use all of these anymore.

Reply:
yeah, it's probably best to remove the stubs, especially from this namespace. it won't break the API if/when we add the functions back upon implementation.